send.c

```diff
@@ -652,9 +652,14 @@ static struct safe_buffer_list *add_ra_options_dnssl(struct safe_buffer_list *sb
 		serialized_domains->used = 0;
 		size_t const domain_name_bytes = serialize_domain_names(serialized_domains, dnssl);
 		size_t const bytes = sizeof(dnsslinfo) + domain_name_bytes;
-		if (bytes > (256 * 8)) {
-			flog(LOG_ERR, "DNSSL too long for RA option, must be < 2048 bytes.  Exiting.");
-			exit(1);
+		// dnsslinfo.nd_opt_dnssli_len is uint8 count of 8-octet groups; min 3, max 255
+		// too many long serialized domains could exceed it
+		// https://datatracker.ietf.org/doc/html/rfc8106#section-5.2
+		size_t const bytes_rounded_up = (bytes + 7) / 8;
+		if(bytes_rounded_up > (255*8)) {
+			flog(LOG_ERR, "DNSSL too long for RA option, must be < 2032 bytes. Skipping option.");
```

Comment:
I'm confused about the math here. `255*8` is `2040`. So should this error message read `<= 2040 bytes`? Or did I miss something?

Reply:
A lot of magic numbers here. I suggest adding macro constants to document what they mean.

Reply:
2032 bytes of suffixes, plus 8 bytes of header = 2040; will introduce macros for clarity.

But I also think there is a subtle bug here with very large configs that needs fixing

Reply:
@mpontillo  this should be much clearer now - I also added testcases for the edges involved